File: 2406-decode-the-message/decode-the-message.js
var decodeMessage = function(key, message) {
    key = key.replaceAll(" ", "");
    let hash = {};
    let ask = 97;
    let i = 0;
    let Output = "";

    while (ask < 123 && i < key.length){
        let char = String.fromCharCode(ask);
        if (!(key[i] in hash)){
            hash[key[i]] = char;
            ask++;
        }
        i++;
    }

    for (let each of message){
        if (each == " "){
            Output += " ";
        }
        else{
            Output += hash[each];
        }
    }
    return Output; 
    
};

File: 2406-decode-the-message/decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        Output = ""
        key = key.replace(" ","")
        ASCII = 97
        i = 0
        hash = {}
        while ASCII < 123 and i < len(key):
            if key[i] not in hash:
                char = chr(ASCII)
                hash[key[i]] = char
                ASCII += 1
            i += 1

        for i in range(0,len(message)):
            
                if message[i] == " ":
                    Output += " "
                else: 
                    Output += hash[message[i]]
        
        return Output